### ISSUE TITLE
Clarification on first test example conditioning on data from previous year, and test data stride

### ISSUE THREAD

**Issue by mjwillson** [pangeo-data/WeatherBench#38]
Hello! I'm replicating the evaluation for WeatherBench and have some (admittedly slightly pedantic) clarification requests. The WeatherBench paper says:

"To make sure no overlap exists between the training and test dataset, the first test date is 1 January 2017 00UTC plus forecast time (i.e. for a three day forecast the first test date would be 4 January 2017 00UTC)".

But in Rasp & Thuerey 2020 at least, your CNN conditions on inputs at t=0, -6, -12hr relative to the reference time. Does that mean the first test example conditions on inputs from the end of the training/validation set? (2016-12-31 18:00, 2016-12-31 12:00 at t-6, t-12 respectively) ?

I don't think this is a problem, but it would be good to be clear about it if this is what you're doing. I imagine most people would generate test examples after making a clean division of train / valid / test sets without any overlap, but in that case the first test example above wouldn't be available, and the evaluation wouldn't be on the exact same set of test examples as the baselines in the paper.

For similar reasons (wanting to replicate the same set of test examples) it would also be good to state explicitly what stride to use when generating test examples, as part of the benchmark. (I think I saw a 6 hour stride in your code?)

Thanks!

**Comment by raspstephan** [comment]
First of all, thanks for the question! 

If I remember correctly (90% sure) the first input sample (-12h) will be 2017-01-01 00:00, so that absolute no overlap exists. So yes, in fact the first prediction is actually later, so for a 3 day prediction: 3 days + 12h. You are right that this isn't entirely clear from the paper.

And yes, the stride is 6 hours, sorry for not mentioning this more clearly. 

Having said all this, I don't think it will make a noticeable difference but it's better to be exact for sure.

**Comment by mjwillson** [comment]
Aha, OK thanks! That keeps it simple then.

You're right it probably doesn't make a big difference to the numbers, but it is one extra source of variance when comparing models which it's nice to remove if it's easy to do.

**Comment by codenust** [comment]
@mjwillson How can we check the prediction on a specific day in 2017? The CNN predictions are calculated for the entire 2017 by and 2018 by fc = create_predictions(cnn, dg_test) so for any given day in 2017, say June 1st, if we plot the fc on that day what would that mean? is it the predicted data coming from the inputs 3 days ago? or do we have to extend the training data to May 31st?
Thank you
